storagemarket/provider_test.go

```diff
@@ -0,0 +1,552 @@
+package storagemarket
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"math/rand"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/filecoin-project/boost/storagemarket/smtestutil"
+
+	"github.com/libp2p/go-libp2p-core/event"
+
+	carv2 "github.com/ipld/go-car/v2"
+
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/boost/storagemarket/types/dealcheckpoints"
+
+	"github.com/filecoin-project/go-state-types/big"
+
+	"github.com/filecoin-project/lotus/api"
+
+	types2 "github.com/filecoin-project/boost/transport/types"
+
+	acrypto "github.com/filecoin-project/go-state-types/crypto"
+
+	"github.com/filecoin-project/boost/testutil"
+	"github.com/google/uuid"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+
+	"github.com/filecoin-project/boost/db"
+	"github.com/filecoin-project/boost/fundmanager"
+	"github.com/filecoin-project/boost/storagemanager"
+	"github.com/filecoin-project/boost/storagemarket/types"
+	"github.com/filecoin-project/go-address"
+	lotusmocks "github.com/filecoin-project/lotus/api/mocks"
+
+	"testing"
+
+	"github.com/filecoin-project/lotus/node/repo"
+	"github.com/golang/mock/gomock"
+	"github.com/libp2p/go-libp2p-core/peer"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSimpleDealHappy(t *testing.T) {
+	ctx := context.Background()
+	testMode = true
+	defer func() {
+		testMode = false
+	}()
+
+	// setup the provider test harness
+	// TODO Do not hardcode these values
+	minPublish := abi.NewTokenAmount(100)
+	harness := NewHarness(t, ctx, 10000000000, minPublish)
+	defer harness.Stop()
+	// start the provider test harness
+	harness.Start(t, ctx)
+
+	// build the deal proposal
+	dp, carV2FilePath := harness.mkDealProposal(t, harness.NormalServer.URL, 1)
+
+	// setup mock publish & add-piece expectations
+	so := harness.MinerStub.ForDeal(dp).SetupAllBlocking().Output()
+	harness.setupWalletBalances(t, big.NewInt(300), big.NewInt(500), 1000)
```

Comment:
I suggest having harness defaults for the wallet balances that can be overridden with options

Reply:
I want these values to be explicit, so people think about it when writing tests. Keeping this hidden will then make you spend time debugging tests wondering why things are failing when you have too many deals.

Reply:
Ok makes sense